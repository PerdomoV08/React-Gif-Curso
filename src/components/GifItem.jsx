import PropTypes from 'prop-types';
import checkPropTypes from 'prop-types/checkPropTypes';
import '@testing-library/jest-dom';

const GifItem = (props) => {
    // Validación manual de PropTypes (necesaria en React 19 para componentes funcionales)
    checkPropTypes(GifItem.propTypes, props, 'prop', 'GifItem');

    const { title, url, id } = props;

    return (
        <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
        </div>
    );
};

// Definición de PropTypes
GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string
};

export default GifItem
