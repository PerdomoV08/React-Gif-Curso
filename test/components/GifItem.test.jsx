// @ts-nocheck
import {render, screen} from '@testing-library/react'
import GifItem from '../../src/components/GifItem'

describe('Pruebas en el componente <GifItem />', () => {

    const title = 'HxH';
    const url = 'htts://hunterXhunter.com';

    test('Debe de hacer macth con el snapshot', () => {
        const {container} = render(<GifItem title = {title} url = {url}/>)
        expect(container).toMatchSnapshot();
    })
    
    test('Debe de mostrar la imagen con el URL y ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        //screen.debug() // Muestra una ayuda para comprender mejor su funcionamiento
        // expect(screen.getByRole('img').src).toBe(url) // Otra forma de hacerlo
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    })
    
    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url= {url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
    

})
