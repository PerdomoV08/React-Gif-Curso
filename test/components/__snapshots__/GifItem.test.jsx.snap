// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem /> Debe de hacer macth con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="HxH"
      src="htts://hunterXhunter.com"
    />
    <p>
      HxH
    </p>
  </div>
</div>
`;
